payments-core/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -409,6 +428,10 @@ internal class DefaultFlowController @Inject internal constructor(
         }
     }
 
+    protected fun finalize() {
+        paymentController.unregisterLaunchers()
+    }
```

Comment:
I don't think we should rely on `finalize()`

https://developer.android.com/reference/java/lang/Object.html#finalize%28%29

Can we hook into the lifecycle methods?

Reply:
Done with some changes to default `FlowControllerFactory` interface